apps/studio/components/interfaces/Database/Functions/CreateFunction.tsx

```diff
@@ -483,409 +413,192 @@ const InputMultiArguments = observer(({ readonly }: InputMultiArgumentsProps) =>
         </p>
       </div>
       <div className="space-y-2 pt-4">
-        {readonly && isEmpty(_localState!.formState.args.value) && (
+        {readonly && isEmpty(fields) && (
           <span className="text-foreground-lighter">No argument for this function</span>
         )}
-        {_localState!.formState.args.value.map(
-          (x: { name: string; type: string; error?: string }, idx: number) => (
-            <InputArgument
-              key={`arg-${idx}`}
-              idx={idx}
-              name={x.name}
-              type={x.type}
-              error={x.error}
-              readonly={readonly}
-            />
+        {fields.map((field, index) => {
+          return (
+            <div className="flex flex-row space-x-1" key={field.id}>
+              <FormField_Shadcn_
+                name={`args.${index}.name`}
+                render={({ field }) => (
+                  <FormItemLayout className="flex-1">
+                    <FormControl_Shadcn_>
+                      <Input_Shadcn_ {...field} readOnly disabled={readonly} />
+                    </FormControl_Shadcn_>
+                  </FormItemLayout>
+                )}
+              />
+              <FormField_Shadcn_
+                name={`args.${index}.type`}
+                render={({ field }) => (
+                  <FormItemLayout className="flex-1">
```

Comment:
probably shouldn't use FormItemLayout here. it serves no purpose.

this layout is unique
<img width="632" alt="Screenshot 2024-03-11 at 1 15 43 PM" src="https://github.com/supabase/supabase/assets/8291514/d4578be9-cab7-4653-8294-690c8f5e2e1f">